tests/components/hassio/test_backup.py

```diff
@@ -231,6 +231,67 @@ async def test_agent_delete_backup(
     supervisor_client.backups.remove_backup.assert_called_once_with(backup_id)
 
 
+@pytest.mark.usefixtures("hassio_client")
+@pytest.mark.parametrize(
+    ("event_data", "mount_info_calls"),
+    [
+        (
+            {
+                "event": "job",
+                "data": {"name": "mount_manager_create_mount", "done": True},
+            },
+            1,
+        ),
+        (
+            {
+                "event": "job",
+                "data": {"name": "mount_manager_create_mount", "done": False},
+            },
+            0,
+        ),
+        (
+            {
+                "event": "job",
+                "data": {"name": "mount_manager_remove_mount", "done": True},
+            },
+            1,
+        ),
+        (
+            {
+                "event": "job",
+                "data": {"name": "mount_manager_remove_mount", "done": False},
+            },
+            0,
+        ),
+        ({"event": "job", "data": {"name": "other_job", "done": True}}, 0),
+        (
+            {
+                "event": "other_event",
+                "data": {"name": "mount_manager_remove_mount", "done": True},
+            },
+            0,
+        ),
+    ],
+)
+async def test_agents_notify_on_mount_added_removed(
+    hass: HomeAssistant,
+    hass_ws_client: WebSocketGenerator,
+    supervisor_client: AsyncMock,
+    event_data: dict[str, Any],
+    mount_info_calls: int,
+) -> None:
+    """Test the listener is called when mounts are added or removed."""
+    client = await hass_ws_client(hass)
+    supervisor_client.mounts.info.assert_called_once_with()
```

Comment:
I'd not use assertion methods on this mock since the `supervisor_client` mock isn't specced. There's a risk that a type passes unnoticed. I recommend using `Mock.call_count` and `Mock.call_args`.